shrinkwrap/elf.py

```diff
@@ -0,0 +1,50 @@
+from __future__ import annotations
+
+import re
+from abc import ABC, abstractmethod
+from typing import Dict
+
+import lief  # type: ignore
+from sh import Command  # type: ignore
+
+
+class LinkStrategy(ABC):
+    @staticmethod
+    def select_by_name(name: str) -> LinkStrategy:
+        if name == "native":
+            return NativeLinkStrategy()
+        elif name == "virtual":
+            return VirtualLinkStrategy()
+        else:
+            raise Exception(f"Unknown strategy: {name}")
+
+    @abstractmethod
+    def explore(self, binary: lief.Binary, filename: str) -> Dict[str, str]:
+        """
+        Determine the linking for all needed objects
+        """
+        pass
+
+
+class NativeLinkStrategy(LinkStrategy):
+    """Uses the native interpreter in the binary to determine the linking"""
+
+    def explore(self, binary: lief.Binary, filename: str) -> Dict[str, str]:
```

Comment:
kind of annoying that the binary doesn't give the original filename and just the name portion :/
Could move a bunch of the lief usage into an ELF friendly class maybe in the future.